src/omlt/linear_tree/lt_definition.py

```diff
@@ -0,0 +1,408 @@
+import numpy as np
+import lineartree
+
+
+class LinearTreeDefinition:
+    """
+    Class to represent a linear tree model trained in the linear-tree package
+
+    Attributes:
+        __model (linear-tree model) : Linear Tree Model trained in linear-tree
+        __splits (dict) : Dict containing split node information
+        __leaves (dict) : Dict containing leaf node information
+        __thresholds (dict) : Dict containing splitting threshold information
+        __scaling_object (scaling object) : Scaling object to ensure scaled
+            data match units of broader optimization problem
+        __scaled_input_bounds (dict): Dict containing scaled input bounds
+        __unscaled_input_bounds (dict): Dict containing unscaled input bounds
+
+    References:
+        * linear-tree : https://github.com/cerlymarco/linear-tree
+        * Ammari et al. (2023) Linear Model Decision Trees as Surrogates in
+            Optimization of Engineering Applications. Computers & Chemical
+            Engineering
+        * Misic, V. "Optimization of tree ensembles."
+             Operations Research 68.5 (2020): 1605-1624.
+        * Mistry, M., et al. "Mixed-integer convex nonlinear optimization
+             with gradient-boosted trees embedded." INFORMS Journal on
+             Computing (2020).
+    """
+
+    def __init__(
+        self,
+        lt_regressor,
+        scaling_object=None,
+        scaled_input_bounds=None,
+        unscaled_input_bounds=None,
+    ):
+        """Create a LinearTreeDefinition object and define attributes based on the
+        trained linear model decision tree.
+
+        Arguments:
+            lt_regressor -- A LinearTreeRegressor model that is trained by the
+                linear-tree package
+
+        Keyword Arguments:
+            scaling_object --A scaling object to specify the scaling parameters
+                for the linear model tree inputs and outputs. If None, then no
+                scaling is performed. (default: {None})
+            scaled_input_bounds -- A dict that contains the bounds on the scaled
+                variables (the direct inputs to the tree). If None, then the
+                user must specify the bounds via the input_bounds argument.
+                (default: {None})
+            unscaled_input_bounds -- A dict that contains the bounds on the
+                variables (the direct inputs to the tree). If None, then the
+                user must specify the scaled bounds via the scaled_input_bounds
+                argument. (default: {None})
+
+        Raises:
+            Exception: Input bounds required. If unscaled_input_bounds and
+                scaled_input_bounds is None, raise Exception.
+        """
```

Comment:
It looks like a different doc-string format than I am used to. Can you verify that these work in the API docs?

Reply:
I generated the API docs and this format is fine. Note that I also added an omlt.linear_tree.rst file in docs\api_docs\ to generate the API docs using tox.